components/table/table-cell.tsx

```diff
@@ -20,8 +21,11 @@ export type cellData = {
 }
 
 const TableCell: React.FC<Props> = ({ columns, row, rowIndex, emptyText, onCellClick }) => {
-  const { removeRow } = useTableContext()
+  const { removeRow, updateRow } = useTableContext()
   const actions: cellActions = {
+    update: (data: any) => {
```

Comment:
Can we infer the parameter types here?

Reply:
not sure, didn't do much typescript apart from basic in a while, so you mean a generic? because in the types, i saw that data was an `any[]` so i assumed we didn't the type of what's inside data. happy to infer if you think it's possible somehow.

Reply:
We should be able to infer the type (I'm not sure, you can try it in the code first). 
Besides, are you sure that the `data` parameter in `updateRow` function is the same thing as `dataRef`?  Each entry in the `dataRef` array appears to be of type `Array<TableColumnItem>`, state `dataRef` seems to represent every column after abstraction, I'm not sure if this update will work properly, you can try printing `dataRef`, yhey may not be the same data.

Reply:
i just check for `dataRef` and it's the right one, it's also the one used in `removeRow`.

Reply:
for the types, i did this but not sure this is what you meant:
```
    update: function <T>(data: T) {
      updateRow && updateRow(rowIndex, data)
    },
```
from what i understand, to infer i would need to leave it without a type and let the checker guess the type when people pass their data but then i get an error in the editor saying `data` has an implicit any.